src/string_util.cc

```diff
@@ -92,21 +91,24 @@ std::string ExponentToPrefix(int64_t exponent, bool iec) {
   const int64_t index = (exponent > 0 ? exponent - 1 : -exponent - 1);
   if (index >= kUnitsSize) return "";
 
-  const char* array =
+  const std::string* array =
       (exponent > 0 ? (iec ? kBigIECUnits : kBigSIUnits) : kSmallSIUnits);
   if (iec) {
-    return array[index] + std::string("i");
+    return array[index] + "i";
   }
-  return std::string(1, array[index]);
+  return array[index];
 }
 
 std::string ToBinaryStringFullySpecified(double value, double threshold,
-                                         int precision, double one_k = 1024.0) {
+                                         int precision, benchmark::Counter::OneK one_k = benchmark::Counter::kIs1024) {
```

Comment:
depending on `Counter` isn't ideal as it creates a circular dependency, i think.  can we extract the `OneK` enum to "string_util.h" instead?

Reply:
So, I have to -

```cpp
class Counter {
 public:
  enum Flags {
    //flags
  };
// ... removed the enum OneK
};

```
and then place this in `string_util.h` like the below?

```cpp
namespace benchmark {
enum OneK {
  kIs1000 = 1000,
  kIs1024 = 1024
};
} 
```

I also missed this update in the latest commit, sending this change after your confirmation.

Reply:
I made the changes to avoid the circular dependency and corrected the rest of the code according to the changes and created a new commit for these changes.

Also at line 849 in file benchmark.h. Updated the below function.

```cpp
  BENCHMARK_ALWAYS_INLINE
  void SetBytesProcessed(int64_t bytes) {
    counters["bytes_per_second"] =
        Counter(static_cast<double>(bytes), Counter::kIsRate, kIs1024);
  }
```